include/xtensor-python/pycontainer.hpp

```diff
@@ -321,6 +321,30 @@ namespace xt
         return *static_cast<const derived_type*>(this);
     }
 
+    namespace detail
+    {
+        template <class S>
+        struct check_dims
+        {
+            bool operator()(std::size_t)
+            {
+                return true;
+            }
+        };
+
+        template <class T, std::size_t N>
+        struct check_dims<std::array<T, N>>
+        {
+            bool operator()(std::size_t new_dim)
```

Comment:
maybe these could be static.

Reply:
it's a static `::run(size_t)` method now